crates/server/src/api/v1/package.rs

```diff
@@ -85,6 +95,33 @@ impl Config {
     }
 }
 
+/// Represents a valid package name.
+///
+/// Valid package names conform to the component model specification
+/// for identifiers.
+///
+/// A valid component model identifier is the format `<namespace>:<name>`,
+/// where both parts are also kebab-case WIT identifiers.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+struct PackageName<'a> {
+    namespace: &'a str,
+    name: &'a str,
+}
```

Comment:
I think this belongs in the protocol crate next to `LogId` and we'll want to make `LogId` derivable from this.